situp-core/src/main/java/com/amazon/situp/parser/PipelineParser.java

```diff
@@ -113,9 +126,12 @@ private int getDefaultProcessorThreads() {
             final PluginSetting pluginSetting,
             final Map<String, Pipeline> pipelineMap,
             final Map<String, PipelineConfiguration> pipelineConfigurationMap) {
+        LOG.info("Building [{}] as source component for the pipeline [{}]", pluginSetting.getName(), sourcePipelineName);
         final Optional<String> pipelineNameOptional = getPipelineNameIfPipelineType(pluginSetting);
```

Comment:
Pipeline.Name has to be present we are using PipelineConnector right? Given that we handle the exceptions, we can now avoid the Optional and throw exception.

Reply:
if the configured source is of type pipeline, then it will return pipeline name; if it is not of pipeline type, then we won't have name. so i think we need Optional.

Reply:
My bad, this makes sense.